eclair-core/src/main/scala/fr/acinq/eclair/Setup.scala

```diff
@@ -353,12 +352,47 @@ class Setup(datadir: File,
       None
     }
   }
+
+  private def initDatabase(dbConfig: Config): Databases = {
+    db match {
+      case Some(d) => d
+      case None =>
+        dbConfig.getString("driver") match {
+          case "sqlite" => Databases.sqliteJDBC(chaindir)
+          case "psql" =>
+            val psql = Databases.setupPsqlDatabases(dbConfig, { ex =>
+              logger.error("fatal error: Cannot obtain lock on the database.\n", ex)
+              sys.exit(-2)
+            })
+            if (LockType(dbConfig.getString("psql.lock-type")) == LockType.EXCLUSIVE) {
```

Comment:
Is `optimistic` locking not implemented?

Reply:
It is.